### PR TITLE
Remove hover props from Pressability

### PR DESCRIPTION
**Pull request by necolas**
## Summary

This patch removes code in `Pressability` that is unused and not backed by events in React Native.

## Test Plan

CI. Should be no change as this functionality is a noop on native, and the associated unit tests have been removed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -77,16 +77,18 @@ Code-level contributions to React Native generally come in the form of [pull req
...
 1. Fork the React Native repository and create your branch from `master`.
-2. Make the desired changes to React Native sources. Use the `RNTester` app to test them out.
-3. If you've added code that should be tested, add tests.
-4. If you've changed APIs, update the documentation, which lives in [another repo](https://github.com/facebook/react-native-website/).
-5. Ensure the test suite passes, either locally or on CI once you opened a pull request.
-6. Make sure your code lints (for example via `yarn lint --fix`).
-7. Push the changes to your fork.
-8. Create a pull request to the React Native repository.
-9. Review and address comments on your pull request.
+2. Run `yarn` to install dependencies.
```
Comment (TheSavior) [inline CONTRIBUTING.md:80]: If you number these all with `1.` then markdown will auto number them and future diffs will be simpler when adding steps

**Comment by acoates-ms** [comment]
What is the plan for hover / focus with pressability (since that is functionality we will want on various RN platforms)

**Comment by tom-un** [comment]
Besides needing hover / focus in the out-of-tree platforms like macOS and Windows, iPadOS just added mouse and hover events.   So core is going to need these events for iOS too.

**Comment by necolas** [comment]
1. This is an internal module that isn't part of the public API
2. `Pressable` already has no support for these props
3. Hover and focus interactions are not part of the pointer responder system, i.e., not something a press responder should be incorporating, but something Pressable should.
4. Events in general need rethinking in RN. The current `onClick` prop is non-standard and incorrectly typed. Other input types should be supported by the addition of W3C keyboard and pointer events

The implementation of React Native for Web has focus events for Pressable and converts mouse events into touch events within the responder system.

I'd like to see touch and mouse events replaced with pointer events rather than adding mouse events to RN, and the responder system events modified to reference "pointers" rather than "touches". There are also many events that aren't touch events that need to be accounted for by the responder system, which means that native events attached to responder events can't only be typed as touch events.

We'd still want a higher level abstraction over pointer events that is somewhat like touch events in that it tracks all the currently active and changed pointers. I've got examples of what all these API changes would look like but it's not really the concern of this PR. Happy to share if it helps to align all our RN forks on W3C standards until a time when mainline RN can assess potential changes

**Comment by necolas** [comment]
Is it expected that several CI checks fail on PRs at the moment?

**Comment by TheSavior** [comment]
The failures all look real and are relevant to this diff

**Comment by necolas** [comment]
Please can you help me understand what in this diff is causing all these errors in `test_js`?

```
/home/circleci/react-native/bots/make-comment.js
  39:13  warning  'body' is already declared in the upper scope  no-shadow

/home/circleci/react-native/IntegrationTests/AppEventsTest.js
  41:5  warning  Do not use setState in componentDidMount  react/no-did-mount-set-state

/home/circleci/react-native/Libraries/Animated/src/__tests__/Animated-test.js
  141:5  warning  Skipped test  jest/no-disabled-tests

/home/circleci/react-native/Libraries/BatchedBridge/__tests__/NativeModules-test.js
  17:5  warning  'fs' is assigned a value but never used               no-unused-vars
  18:5  warning  'parseErrorStack' is assigned a value but never used  no-unused-vars

/home/circleci/react-native/Libraries/Components/Keyboard/Keyboard.js
  114:7  warning  'Keyboard' is assigned a value but never used  no-unused-vars

/home/circleci/react-native/Libraries/Components/ScrollResponder.js
  490:11  warning  Designated alias 'that' is not assigned to 'this'  consistent-this
  514:11  warning  Designated alias 'that' is not assigned to 'this'  consistent-this
  569:11  warning  Designated alias 'that' is not assigned to 'this'  consistent-this

/home/circleci/react-native/Libraries/Components/TextInput/AndroidTextInputNativeComponent.js
  28:13  warning  'React' is defined but never used  no-unused-vars

/home/circleci/react-native/Libraries/Components/TextInput/RCTMultilineTextInputNativeComponent.js
  16:14  warning  'Int32' is defined but never used  no-unused-vars
  18:13  warning  'React' is defined but never used  no-unused-vars

/home/circleci/react-native/Libraries/Components/TextInput/RCTSingelineTextInputNativeComponent.js
  16:14  warning  'Int32' is defined but never used  no-unused-vars
  17:13  warning  'React' is defined but never used  no-unused-vars

/home/circleci/react-native/Libraries/Components/TextInput/TextInput.js
  1000:11  warning  'text' is already declared in the upper scope  no-shadow

/home/circleci/react-native/Libraries/Components/Touchable/PooledClass.js
  23:9  warning  Unexpected alias 'Klass' for 'this'  consistent-this
  34:9  warning  Unexpected alias 'Klass' for 'this'  consistent-this
  45:9  warning  Unexpected alias 'Klass' for 'this'  consistent-this
  56:9  warning  Unexpected alias 'Klass' for 'this'  consistent-this
  67:9  warning  Unexpected alias 'Klass' for 'this'  consistent-this

/home/circleci/react-native/Libraries/LogBox/UI/AnsiHighlight.js
  80:18  warning  Inline style: { flexDirection: 'column' }  react-native/no-inline-styles
  82:22  warning  Inline style: { flexDirection: 'row' }     react-native/no-inline-styles

/home/circleci/react-native/Libraries/Pressability/PressabilityDebug.js
  58:18  warning  Inline style: { borderStyle: 'dashed', borderWidth: 1, position: 'absolute' }  react-native/no-inline-styles

/home/circleci/react-native/ReactAndroid/src/androidTest/js/MeasureLayoutTestModule.js
  50:24  warning  Using this.refs is deprecated                          react/no-string-refs
  51:24  warning  Using this.refs is deprecated                          react/no-string-refs
  52:24  warning  Using this.refs is deprecated                          react/no-string-refs
  53:24  warning  Using this.refs is deprecated                          react/no-string-refs
  58:13  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  59:15  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  60:17  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  62:15  warning  Using string literals in ref attributes is deprecated  react/no-string-refs

/home/circleci/react-native/ReactAndroid/src/androidTest/js/PickerAndroidTestModule.js
  29:5  warning  Unexpected alias 'appInstance' for 'this'  consistent-this

/home/circleci/react-native/ReactAndroid/src/androidTest/js/ScrollViewTestModule.js
  108:5  warning  Unexpected alias 'scrollViewApp' for 'this'  consistent-this
  142:5  warning  Unexpected alias 'scrollViewApp' for 'this'  consistent-this

/home/circleci/react-native/ReactAndroid/src/androidTest/js/SubviewsClippingTestModule.js
  330:5  warning  Unexpected alias 'appInstance' for 'this'  consistent-this

/home/circleci/react-native/ReactAndroid/src/androidTest/js/SwipeRefreshLayoutTestModule.js
  54:5  warning  Unexpected alias 'app' for 'this'  consistent-this

/home/circleci/react-native/ReactAndroid/src/androidTest/js/TextInputTestModule.js
   73:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
   89:5   warning  Unexpected alias 'app' for 'this'                      consistent-this
  121:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  127:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  132:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  138:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  144:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs

/home/circleci/react-native/RNTester/js/examples/Layout/LayoutEventsExample.js
   99:15  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  101:13  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  115:17  warning  Using string literals in ref attributes is deprecated  react/no-string-refs

/home/circleci/react-native/RNTester/js/examples/NativeAnimation/NativeAnimationsExample.js
   66:19  warning  Script URL is a form of eval  no-script-url
  344:19  warning  Script URL is a form of eval  no-script-url

/home/circleci/react-native/RNTester/js/examples/RTL/RTLExample.js
  30:6  warning  'State' is defined but never used  no-unused-vars

/home/circleci/react-native/RNTester/js/examples/TextInput/TextInputSharedExamples.js
  187:5   warning  Using this.refs is deprecated                          react/no-string-refs
  194:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  202:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  211:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  220:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs
  228:11  warning  Using string literals in ref attributes is deprecated  react/no-string-refs

/home/circleci/react-native/RNTester/js/RNTesterApp.ios.js
  149:43  warning  Expected error to be handled  handle-callback-err

/home/circleci/react-native/scripts/android-e2e-test.js
  97:9  warning  Expected error to be handled  handle-callback-err

✖ 57 problems (0 errors, 57 warnings)

Done in 25.58s.
```

**Comment by necolas** [comment]
I was also told the Touchables don't support blur/focus, but it looks like they do.

**Comment by TheSavior** [comment]
Those lint warnings are unrelated and not causing the builds to fail. Those are okay to ignore in this PR

**Comment by necolas** [comment]
Thanks. Looks like everything is passing now

**Comment by yungsters** [comment]
> I was also told the Touchables don't support blur/focus, but it looks like they do.

To clarify the context that I had shared, React Native does not broadly support blur and focus events (outside of TextInput). The Touchable components support props named `onBlur` and `onFocus`, but they are implemented using `TVTouchable` (and only applicable if `Platform.isTV`).

Is this the behavior that you discovered? Or are you referring to something completely different?

Nothing that I'm aware of is relying on the `onFocus` and `onBlur` configuration properties in `Pressability`. In fact, if you grep for `eventHandlersWithoutBlurAndFocus`, you'll see that I had to intentionally omit these event handlers from components like `TouchableWithoutFeedback` because it was inadvertently overriding the `onFocus` of `TextInput`. (Yes, `TouchableWithoutFeedback` is currently an implementation detail of `TextInput`.)

**Comment by charlesbdudley** [comment]
Hi, @necolas, I'm going to close this for now so we can focus on active PRs. Feel free to reopen if you believe this is in error.

**Pull request closed without merging**